.github/src/EngineManager.java

```diff
@@ -1,15 +1,73 @@
 /*
- * represented by "Comp #1" in system diagram
+ * Assignment 4 instructions: 
+ * In the 'src' folder, implement the coordination component of the engine.
+ * This piece will do a couple of things, although exactly how it does these is
+ * up to you!
+ *      a.Receive requests from the user to start the computation (the
+ *        method signature for this will come from your API), and return a
+ *        suitable result status back to the user
+ * 
+ *      b. Request that the data storage component read integers from the
+ *      location specified by the user (remember, the number of integers
+ *      involved is potentially very large!)
+ * 
+ *      c. Pass the integers to the compute component
+ * 
+ *      d. Request that the data storage component write the results to the output
  */
 
+import java.util.List;
+
 public class EngineManager implements ComputeAPI {
+
     DataAPI dataStoreAPI;
     EngineAPI computeEngineAPI;
 
+    public EngineManager(DataStorage dataStorage, EngineCompute engineCompute ) {
+
+        dataStoreAPI =  dataStorage;
+        computeEngineAPI = engineCompute;
+
+    }
+
+    /*Receive requests from the user to start the computation (the
+    * method signature for this will come from your API), and return a
+    * suitable result status back to the user
+    */
     @Override
     public ComputeResult compute(ComputeRequest request) {
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'compute'");
+        
+        //request DataAPI object to read ints
+        request = new ComputeRequestTestImpl();
+        InputConfig inputConfig = request.getInputConfig();
+        DataReadResult readRequest = dataStoreAPI.read(inputConfig);
```

Comment:
I think that line 42 and 43 could be condensed down to : DataReadResult readRequest = dataStoreAPI.read(request.getInputConfig());

this method call would have the correct return type to pass into the arguments of the "read" method call.